Source/Synthesis/Generation/MetadataGenerator.cs

```diff
@@ -59,7 +60,10 @@ public TemplateGenerationMetadata GenerateMetadata()
 
 						var fieldInfo = new FieldPropertyInfo(field);
 						fieldInfo.FieldPropertyName = propertyName;
-						fieldInfo.SearchFieldName = _indexFieldNameTranslator.GetIndexFieldName(field.Name);
+						if (_indexFieldNameTranslator.GetType().Name.ToLower().Contains("solr"))
+							SolrFieldMappingHack(field, fieldInfo);
+						else
+							fieldInfo.SearchFieldName = _indexFieldNameTranslator.GetIndexFieldName(field.Name);
```

Comment:
can you think of a better way to identify if a solr index is being used?  This seems a little weak to me.  However i wanted to keep away from a config setting.  The less complex it is to set up the better i think.